Embree build failure after pulldown on Windows

### Describe the bug

We get the below CC error when trying to build Embree on Windows.
It's failing after [this](https://github.com/intel/llvm/commit/8bcf1108f7edb2c2ff17535844ca3a9b1d2909c2) pulldown but I don't know for sure that's the cause.
The error itself seems to be from TBB:

```
FAILED: kernels/CMakeFiles/embree.dir/sycl/scene_sycl.cpp.obj 
D:\github\_work\llvm\llvm\install\bin\clang++.exe -DEMBREE_LOWEST_ISA -DEMBREE_SYCL_SUPPORT -DEMBREE_TARGET_AVX -DEMBREE_TARGET_AVX2 -DEMBREE_TARGET_SSE2 -DEMBREE_TARGET_SSE42 -DTASKING_TBB -D__INTEL_LLVM_COMPILER -D__TBB_NO_IMPLICIT_LINKAGE=1 -Dembree_EXPORTS -ID:/github/_work/llvm/llvm/embree-build -ID:/github/_work/llvm/llvm/embree-src/kernels/../include -ID:/github/_work/llvm/llvm/embree-src/kernels/level_zero/.. -isystem D:/github/tbb/include -fno-sycl -Wno-bitwise-instead-of-logical -Wno-pessimizing-move -std=c++17 -isystem "D:/github/_work/llvm/llvm/install/bin/../include/sycl" -isystem "D:/github/_work/llvm/llvm/install/bin/../include/" -std=c++17 -fvisibility=hidden -fvisibility-inlines-hidden -fno-strict-aliasing -fno-tree-vectorize -D_FORTIFY_SOURCE=2 -fsigned-char -Wall -Wformat -Wformat-security -ffp-model=precise -O3 -DNDEBUG -D_DLL -D_MT -Xclang --dependent-lib=msvcrt -msse2 -fsycl -fsycl-unnamed-lambda -Xclang -fsycl-allow-func-ptr -Wno-mismatched-tags -Wno-pessimizing-move -Wno-reorder -Wno-unneeded-internal-declaration -Wno-delete-non-abstract-non-virtual-dtor -Wno-dangling-field -Wno-unknown-pragmas -Wno-logical-op-parentheses -g0 -UDEBUG -DNDEBUG -Wno-bitwise-instead-of-logical  -fsycl-targets=spir64 -MD -MT kernels/CMakeFiles/embree.dir/sycl/scene_sycl.cpp.obj -MF kernels\CMakeFiles\embree.dir\sycl\scene_sycl.cpp.obj.d -o kernels/CMakeFiles/embree.dir/sycl/scene_sycl.cpp.obj -c D:/github/_work/llvm/llvm/embree-src/kernels/sycl/scene_sycl.cpp
In file included from D:/github/_work/llvm/llvm/embree-src/kernels/sycl/scene_sycl.cpp:13:
In file included from D:/github/_work/llvm/llvm/embree-src/kernels/sycl\../../common/algorithms/parallel_for.h:6:
In file included from D:/github/_work/llvm/llvm/embree-src/kernels/sycl\../../common/algorithms\../tasking/taskscheduler.h:9:
In file included from D:/github/_work/llvm/llvm/embree-src/kernels/sycl\../../common/algorithms\../tasking\taskschedulertbb.h:31:
In file included from D:/github/tbb/include\tbb/tbb.h:17:
In file included from D:/github/tbb/include\tbb\../oneapi/tbb.h:33:
In file included from D:/github/tbb/include\oneapi/tbb/combinable.h:22:
In file included from D:/github/tbb/include\oneapi/tbb\enumerable_thread_specific.h:26:
In file included from D:/github/tbb/include\oneapi/tbb\concurrent_vector.h:24:
In file included from D:/github/tbb/include\oneapi/tbb\detail/_segment_table.h:25:
D:/github/tbb/include\oneapi/tbb\detail\_exception.h:53:39: error: virtual function 'what' has different calling convention attributes ('const char *() const __attribute__((thiscall))') than the function it overrides (which has calling convention 'const char *() const')
   53 |     const char* __TBB_EXPORTED_METHOD what() const noexcept(true) override;
      |                                       ^
C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.44.35207\include\vcruntime_exception.h:94:36: note: overridden virtual function is here
   94 |     _NODISCARD virtual char const* what() const
      |                                    ^
In file included from D:/github/_work/llvm/llvm/embree-src/kernels/sycl/scene_sycl.cpp:13:
In file included from D:/github/_work/llvm/llvm/embree-src/kernels/sycl\../../common/algorithms/parallel_for.h:6:
In file included from D:/github/_work/llvm/llvm/embree-src/kernels/sycl\../../common/algorithms\../tasking/taskscheduler.h:9:
In file included from D:/github/_work/llvm/llvm/embree-src/kernels/sycl\../../common/algorithms\../tasking\taskschedulertbb.h:31:
In file included from D:/github/tbb/include\tbb/tbb.h:17:
In file included from D:/github/tbb/include\tbb\../oneapi/tbb.h:33:
In file included from D:/github/tbb/include\oneapi/tbb/combinable.h:22:
In file included from D:/github/tbb/include\oneapi/tbb\enumerable_thread_specific.h:26:
In file included from D:/github/tbb/include\oneapi/tbb\concurrent_vector.h:24:
In file included from D:/github/tbb/include\oneapi/tbb\detail/_segment_table.h:25:
D:/github/tbb/include\oneapi/tbb\detail\_exception.h:59:39: error: virtual function 'what' has different calling convention attributes ('const char *() const __attribute__((thiscall))') than the function it overrides (which has calling convention 'const char *() const')
   59 |     const char* __TBB_EXPORTED_METHOD what() const noexcept(true) override;
      |                                       ^
C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.44.35207\include\vcruntime_exception.h:94:36: note: overridden virtual function is here
   94 |     _NODISCARD virtual char const* what() const
      |                                    ^
In file included from D:/github/_work/llvm/llvm/embree-src/kernels/sycl/scene_sycl.cpp:13:
In file included from D:/github/_work/llvm/llvm/embree-src/kernels/sycl\../../common/algorithms/parallel_for.h:6:
In file included from D:/github/_work/llvm/llvm/embree-src/kernels/sycl\../../common/algorithms\../tasking/taskscheduler.h:9:
In file included from D:/github/_work/llvm/llvm/embree-src/kernels/sycl\../../common/algorithms\../tasking\taskschedulertbb.h:31:
In file included from D:/github/tbb/include\tbb/tbb.h:17:
In file included from D:/github/tbb/include\tbb\../oneapi/tbb.h:33:
In file included from D:/github/tbb/include\oneapi/tbb/combinable.h:22:
In file included from D:/github/tbb/include\oneapi/tbb\enumerable_thread_specific.h:26:
In file included from D:/github/tbb/include\oneapi/tbb\concurrent_vector.h:24:
In file included from D:/github/tbb/include\oneapi/tbb\detail/_segment_table.h:25:
D:/github/tbb/include\oneapi/tbb\detail\_exception.h:65:39: error: virtual function 'what' has different calling convention attributes ('const char *() const __attribute__((thiscall))') than the function it overrides (which has calling convention 'const char *() const')
   65 |     const char* __TBB_EXPORTED_METHOD what() const noexcept(true) override;
      |                                       ^
C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.44.35207\include\vcruntime_exception.h:94:36: note: overridden virtual function is here

   94 |     _NODISCARD virtual char const* what() const

      |                                    ^

3 errors generated.

```

https://github.com/intel/llvm/actions/runs/27183796588/job/80251350108

### To reproduce

1. Include a code snippet that is as short as possible
2. Specify the command which should be used to compile the program
3. Specify the command which should be used to launch the program
4. Indicate what is wrong and what was expected


### Environment

- OS: [e.g Windows/Linux]
- Target device and vendor: [e.g. Intel GPU]
- DPC++ version: [e.g. commit hash or output of `clang++ --version`]
- Dependencies version: [e.g. the output of `sycl-ls --verbose`]


### Additional context

_No response_

--- a/clang/lib/Sema/SemaDeclAttr.cpp
+++ b/clang/lib/Sema/SemaDeclAttr.cpp
@@ -5949,29 +5949,15 @@ bool Sema::CheckCallingConvAttr(const ParsedAttr &Attrs, CallingConv &CC,
       A = HostTI->checkCallingConvention(CC);
     if (A == TargetInfo::CCCR_OK && CheckDevice && DeviceTI)
       A = DeviceTI->checkCallingConvention(CC);
-  } else if (LangOpts.SYCLIsDevice) {
-    // During device compilation, calling conventions that are valid for the
-    // host, for the device, and for both the host and the device may be
-    // encountered. Diagnostics are desired for cases where the calling
-    // convention is not supported by either the host or the device. If Aux is
-    // null (which should rarely be the case), it isn't possible to check
-    // whether the calling convention is supported by the host, so just assume
-    // that it is. If the calling convention is supported for the device, there
-    // is no need to check the host; the device target gets priority since this
-    // check is only performed during device compilation.
+  } else if (LangOpts.SYCLIsDevice && TI.getTriple().isAMDGPU() &&
+             CC == CC_X86VectorCall) {
+    // Assuming SYCL Device AMDGPU CC_X86VectorCall functions are always to be
+    // emitted on the host. The MSVC STL has CC-based specializations so we
+    // cannot change the CC to be the default as that will cause a clash with
+    // another specialization.
     A = TI.checkCallingConvention(CC);
-    if (Aux && A == TargetInfo::CCCR_Warning) {
-      // If the calling convention would provoke a warning for the device, check
-      // the host and preserve the warning only if the calling convention would
-      // provoke an error for the host. Otherwise, assume this calling
-      // convention is only used for host only functions.
+    if (Aux && A != TargetInfo::CCCR_OK)
       A = Aux->checkCallingConvention(CC);
-      if (A == TargetInfo::CCCR_Error)
-        A = TargetInfo::CCCR_Warning;
-    } else if (Aux && A == TargetInfo::CCCR_Error) {
-      // Assume this calling convention is only used for host only functions.
-      A = Aux->checkCallingConvention(CC);
-    }
   } else {
     A = TI.checkCallingConvention(CC);
   }

--- a/clang/test/SemaSYCL/sycl-cconv.cpp
+++ b/clang/test/SemaSYCL/sycl-cconv.cpp
@@ -16,7 +16,7 @@ void bar() {
 }
 
 // Check some weird calling convention that is not supported even by x86_64 aux.
-// no-aux-warning@+1 {{'__swiftasynccall__' calling convention is not supported for this target}}
+// expected-warning@+1 {{'__swiftasynccall__' calling convention is not supported for this target}}
 void __attribute__((__swiftasynccall__)) g(void) {}
 
 template <typename name, typename Func>
@@ -51,6 +51,16 @@ static void callee() noexcept {
 void foo() {
    invoke(callee);
 }
+
+class A {
+public:
+  // expected-warning@+1 {{'thiscall' calling convention is not supported for this target}}
+  virtual void  __attribute__((thiscall)) boo() {}
+};
+
+class B : public A {
+  virtual void boo() override {}
+};
 
 int main() {
   //expected-error@+1 {{SYCL device code does not support variadic functions}}

